src/recordlinker/main.py

```diff
@@ -180,6 +219,51 @@ async def link_record(
             message=f"Could not connect to database: {err}"
         )
 
+@app.post("/link")
+async def link_piirecord(
+    request: Request,
+    input: Annotated[LinkInput, Body()],
+    response: Response,
+    db_session: orm.Session = Depends(get_session),
+) -> LinkResponse:
+    """
+    Compare a PII Reocrd with records in the Master Patient Index (MPI) to
+    check for matches with existing patient records If matches are found,
+    returns the patient and person reference id's
+    """
+    pii_record = input.record
+
+    external_id = input.external_person_id
+    algorithm = algorithm_service.get_algorithm_by_label(db_session, input.algorithm)
+
+    if not algorithm:
+        response.status_code = status.HTTP_422_UNPROCESSABLE_ENTITY
```

Comment:
In the case of an error, we don't need to return a actual payload here, fastapi will allow us to just raise an exception with an optional error message of what went wrong.  Take a look at this example https://fastapi.tiangolo.com/tutorial/handling-errors/#use-httpexception